GCC2D.gd

```diff
@@ -50,40 +50,48 @@ func _unhandled_input(e):
 	elif e is InputEventScreenPinch and zoom_gesture == 1:
 		_zoom(e)
 
+func _get_rotation():
+	if (ignore_rotation):
+		return 0 
+	return rotation
+
 # Given a a position on the camera returns to the corresponding global position
 func camera2global(position):
 	var camera_center = global_position
 	var from_camera_center_pos = position - get_camera_center_offset()
-	return camera_center + (from_camera_center_pos*zoom).rotated(rotation)
+	return camera_center + (from_camera_center_pos/zoom).rotated(_get_rotation())
 
 func _move(event):
-	set_position(position - (event.relative*zoom).rotated(rotation))
+	set_camera_position(position - (event.relative/zoom).rotated(_get_rotation()))
 
 func _zoom(event):
 	var li = event.distance
-	var lf = event.distance + event.relative
+	var lf = event.distance - event.relative
+
 	var zi = zoom.x
-	
 	var zf = (li*zi)/lf
-	if zf == 0: return
 	var zd = zf - zi
 	
 	if zf <= MIN_ZOOM and sign(zd) < 0:
-		zf =MIN_ZOOM
+		zf = MIN_ZOOM
 		zd = zf - zi
 	elif zf >= MAX_ZOOM and sign(zd) > 0:
 		zf = MAX_ZOOM
 		zd = zf - zi
 	
 	var from_camera_center_pos = event.position - get_camera_center_offset()
 	zoom = zf*Vector2.ONE
-	if(!set_position(position - (from_camera_center_pos*zd).rotated(rotation))):
+
+	var relative = (from_camera_center_pos*zd) / (zi*zf) 
+	if(!set_camera_position(position + relative.rotated(_get_rotation()))):
 		zoom = zi*Vector2.ONE
 
 func _rotate(event):
-	var fccp = (event.position - get_camera_center_offset()) # from_camera_center_pos = fccp
+	if ignore_rotation:
+		return
```

Comment:
I added this to prevent position changes when trying to rotate when `ignore_rotation` is true.